src/Sylius/Bundle/ApiBundle/StateProvider/Shop/ChannelProvider.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\StateProvider\Shop;
+
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use Sylius\Component\Core\Model\Channel;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @experimental
+ *
+ * @implements ProviderInterface<ChannelInterface>
+ */
+final class ChannelProvider implements ProviderInterface
+{
+    /**
+     * @throws \Exception
+     */
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): array
+    {
+        Assert::same($operation->getClass(), Channel::class);
+
+        if ($this->isShopGetCollectionOperation($operation, $context) && isset($context['sylius_api_channel'])) {
+            return [$context['sylius_api_channel']];
+        }
+
+        throw new \RuntimeException('Only Shop GET collection operation is supported.');
+    }
+
+    private function isShopGetCollectionOperation(Operation $operation, array $context): bool
+    {
+        return $operation instanceof GetCollection && \str_starts_with($context['operation_name'], '_api_/shop');
```

Comment:
There's a `SectionProvider`. Perhaps we should use it, but I'm unsure if it still works with API Platform 3."